include/natalie/abstract_method_object.hpp

```diff
@@ -21,7 +21,7 @@ class AbstractMethodObject : public Object {
 
     virtual void visit_children(Visitor &visitor) override {
         Object::visit_children(visitor);
-        visitor.visit(m_method);
+        visitor.enqueue(m_method);
```

Comment:
I'm not really sure about this rename. A visitor visits the nodes. That it enqueues the data during the visit is more of an implementation detail, that's now leaking through all of the code.

Another issue: if we want to implement `ObjectSpace.count_objects`, we will probably need another visitor to visit all the nodes just to count them instead of cleaning up. Suddenly that one has to work with `enqueue` too, even if the implementation works without a queue.

Reply:
I'm open to suggestions. Constructive ones.